index.bs

```diff
@@ -562,11 +564,160 @@ let transformer = new TransformStream({
   }
 });
 
-// After this, trackGenerator can be assigned to any sink such as a
-// peer connection, or media element.
-trackProcessor.readable
-    .pipeThrough(transformer)
-    .pipeTo(trackGenerator.writable);
+processor.readable.pipeThrough(transformer).pipeTo(generator.writable);
+const videoBefore = document.getElementById('video-before');
+const videoAfter = document.getElementById('video-after');
+videoBefore.srcObject = stream;
+const streamAfter = new MediaStream([generator]);
+videoAfter.srcObject = streamAfter;
+
+</pre>
+
+The same example using a worker:
+<pre class="example">
+// main.js
+const stream = await getUserMedia({video:true});
+const videoTrack = stream.getVideoTracks()[0];
+const processor = new MediaStreamTrackProcessor({track: videoTrack});
+const generator = new MediaStreamTrackGenerator({kind: 'video'});
+const worker = new Worker('worker.js');
+worker.postMessage(
+  {readable: processor.readable, writable: generator.writable},
+  [processor.readable, generator.writable]);
+const videoBefore = document.getElementById('video-before');
+const videoAfter = document.getElementById('video-after');
+videoBefore.srcObject = stream;
+const streamAfter = new MediaStream([generator]);
+videoAfter.srcObject = streamAfter;
+
+// worker.js
+self.onmessage = async function(e) {
+  const transformer = new TransformStream({
+    async transform(videoFrame, controller) {
+        const facePosition = await detectFace(videoFrame);
+        const newFrame = blurBackground(videoFrame, facePosition);
+        videoFrame.close();
+        controller.enqueue(newFrame);
+    }
+  });
+
+  e.data.readable.pipeThrough(transformer).pipeTo(e.data.writable);
+}
+</pre>
+
+## Multi-source processing
```

Comment:
This example seems to gloss over the issue of timestamp synchronization. In particular, if audio data arrives before the corresponding video data, the code would process the audio using the old video data.

It might work if getSpeechData were async; however, then I would wonder about dropping audio frames during the time that we're awaiting getSpeechData.